File: Java/SETPractice/src/main/java/SOLIDPrinciple/SingleResponsibility/UserService.java
package SOLIDPrinciple.SingleResponsibility;

public class UserService {
    public EmailService emailService;
    public User user;

    public UserService(EmailService emailService) {
        this.emailService = emailService;
    }

    public User RegisterUser(String userName, String password, String email) throws Exception {
        if(!emailService.ValidateEmail(email))
            throw new Exception("Invalid Email Address");
        user = new User(userName, password, email);
        emailService.SendEmail("User Registered Successfully!");
        return user;
    }
}

File: DotNet/SETPractice/SOLIDPrinciples/SingleResponsibility/UserService.cs
namespace SOLIDPrinciples.SingleResponsibility
{
    public class UserService
    {
        public EmailService emailService;
        public User user;
        public UserService(EmailService emailService) 
        {
            this.emailService = emailService;
        }

        public User RegisterUser(string username, string password, string email)
        {
            if (!emailService.ValidateEmail(email))
                throw new Exception("Invalid Email Address");
            user = new User(username, password, email);
            var message = UserEmailBuilder.BuildNewRegistrationMessage(email);
            emailService.SendMessage(message);
            Console.WriteLine("User Registered Successfully!");
            return user;
        }

    }
}
